src/lib.rs

```diff
@@ -88,9 +86,94 @@ pub fn main() {
     });
 }
 
-fn init_logging() {
+pub fn init_logging() {
+    const LOGGING_ENV: &str = "CACHEPOT_LOG";
+
+    use env_logger::fmt::{Color, Style};
+    use log::Level;
+    use std::io::Write;
+
+    /// The available service type that cachepot can run as.
+    #[derive(Copy, Clone)]
+    enum Kind {
+        /// A service that connects a coordinator and a remote worker to execute
+        /// a remote build.
+        DistScheduler,
+        /// A service that's used to directly perform remote sandbox compilation
+        DistWorker,
+        /// A background service used by the cachepot compilation wrapper (client)
+        /// to either re-use local compilation cache or schedule a remote
+        /// compilation via a remote scheduler
+        Coordinator,
+        /// A wrapper that masquerades as a compiler but spawns (or talks to) a
+        /// coordinator to perform the actual compilation locally or offload it
+        /// to a distributed cluster (in both cases we can re-use cached artifacts)
+        Client,
+    }
+
+    impl std::fmt::Display for Kind {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            write!(
+                f,
+                "{}",
+                match self {
+                    Kind::DistScheduler => "dist scheduler",
+                    Kind::DistWorker => "dist worker",
+                    Kind::Coordinator => "coordinator",
+                    Kind::Client => "client",
+                }
+            )
+        }
+    }
+
+    let start_coordinator = env::var("CACHEPOT_START_COORDINATOR").is_ok()
+        || env::args_os().any(|a| a == "--start-coordinator");
+
+    // TODO: That's a rough heuristic - share detection logic from cmdline.rs
+    let kind = match (std::env::args().nth(1).as_deref(), start_coordinator) {
+        (Some("scheduler"), _) => Kind::DistScheduler,
+        (Some("worker"), _) => Kind::DistWorker,
+        (_, true) => Kind::Coordinator,
```

Comment:
Not: Splitting the coordinator from the triple in an if case makes this easier to read imho

Reply:
Fixed in 938054adfb540ce3942a6793ab3e146f50ee42bb